packages/modules/providers/notification-sendgrid/src/services/sendgrid.ts

```diff
@@ -58,14 +58,28 @@ export class SendgridNotificationService extends AbstractNotificationProviderSer
 
     const from = notification.from?.trim() || this.config_.from
 
+    let mailContent
+
+    if ("providerContext" in notification) {
+      mailContent = {
+        subject: notification.providerContext?.subject,
+        html: notification.providerContext?.html,
+      }
+    } else {
+      // we can't mix html and templates for sendgrid
+      mailContent = {
+        templateId: notification.template,
```

Comment:
q: don't you still need the subject?

Reply:
No, the templates in sendgrid already have one